common/static/common/js/discussion/views/discussion_topic_menu_view.js

```diff
@@ -146,27 +149,44 @@
             },
 
             // @TODO move into utils.coffee
+            getNameWidth: function(name) {
+                var test = $('<div>'),
+                    width;
+
+                test.css({
+                    'font-size': this.dropdownButton.css('font-size'),
+                    'opacity': 0,
+                    'position': 'absolute',
+                    'left': -1000,
+                    'top': -1000
+                }).html(name).appendTo(document.body);
+                width = test.width();
+                test.remove();
+                return width;
+            },
+
+            // @TODO move into utils.coffee
             fitName: function(name) {
                 var ellipsisText = gettext('…'),
                     partialName, path, rawName;
 
-                if (name.length < this.maxNameWidth) {
+                if (this.getNameWidth(name) < this.maxNameWidth) {
                     return name;
                 } else {
                     path = _.map(name.split('/'), function(item){
                         return item.replace(/^\s+|\s+$/g, '');
                     });
                     while (path.length > 1) {
                         path.shift();
-                        partialName = ellipsisText + '/' + path.join('/');
-                        if (partialName.length > this.maxNameWidth) {
-                            return partialName;
+                        partialName = ellipsisText + ' / ' + path.join(' / ');
+                        if (this.getNameWidth(partialName) < this.maxNameWidth) {
+                          return partialName;
                         }
                     }
                     rawName = path[0];
                     name = ellipsisText + ' / ' + rawName;
-                    while (name.length > this.maxNameWidth) {
-                      rawName = rawName.slice(0, this.maxNameWidth);
+                    while (this.getNameWidth(name) > this.maxNameWidth) {
```

Comment:
I'm assuming that the bug is that this loop never terminates. Do you know yet what is happening?